nibabel/streamlines/tractogram.py

```diff
@@ -136,6 +162,32 @@ def __setitem__(self, key, value):
 
         self.store[key] = value
 
+    def extend(self, other):
```

Comment:
Can't you just inherit this method?   I guess you'd have make the docstrings and message a bit more generic, but it seems a shame to duplicate the code.